setupmeta/scm.py

```diff
@@ -138,8 +138,11 @@ def parsed_version(text, dirty=None):
     def is_dirty(self):
         """
         :return bool: Is checkout folder self.root currently dirty?
+
+        This checks both the working tree and index, in a single command.
+        Ref: https://stackoverflow.com/a/2659808/15690
         """
-        exitcode = self.get_output("diff", "--quiet", "--ignore-submodules", capture=False)
+        exitcode = self.get_output("diff-index", "--quiet", "--ignore-submodules", "HEAD", capture=False)
```

Comment:
Nice! Also didn't know `diff-index`.

`setup.py version --bump` originally was intended to be able to also automatically commit in-source version bumps (like, if one likes to have `__version__ = "1.2.3"` right there in the source), but I ran into edge cases.

I wanted to allow `version --bump` to auto-commit only changes it would do (but not user changes). Wonder now if `diff-index` would help achieve that.